src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -420,7 +423,7 @@ LOG_SUPER_DEBUG("About to initialise heat");
 LOG_SUPER_DEBUG("Initialised heat");
 
     // Initialize some interpolation tables
-    if(this_spin_temp->first_box || (fabs(initialised_redshift - perturbed_field_redshift) > 0.0001) ) {
+    // if(initialization_required) {
```

Comment:
I think I commented this out because I was having segfault issues -- i.e. the tables weren't being initialized sometimes when they should have been. Since this logic is quite complicated, and to re-initialize is really quick, I thought it's just safer to always do it for now.

Reply:
Yeah, that makes sense, just leave it as you have it then.